test/tools/spec-runner/index.js

```diff
@@ -485,6 +485,22 @@ function validateExpectations(commandEvents, spec, savedSessionData) {
       actualCommand.sort = { [expectedKey]: actualCommand.sort.get(expectedKey) };
     }
 
+    if (expectedCommand.createIndexes) {
+      // TODO(NODE-3235): This is a workaround that works because all indexes in the specs
+      // are objects with one key; ideally we'd want to adjust the spec definitions
+      // to indicate whether order matters for any given key and set general
+      // expectations accordingly
+      for (const [i, dbIndex] of actualCommand.indexes.entries()) {
+        expect(Object.keys(expectedCommand.indexes[i].key)).to.have.lengthOf(1);
+        expect(dbIndex.key).to.be.instanceOf(Map);
+        expect(dbIndex.key.size).to.equal(1);
+      }
+      actualCommand.indexes = actualCommand.indexes.map(dbIndex => ({
+        ...dbIndex,
+        key: Object.fromEntries(dbIndex.key.entries())
```

Comment:
I don't think you need to call `key.entries()` here since you're checking that key is a map instance above, so you should be able to just do `fromEntries(dbIndex.key)`

Reply:
done!